### ISSUE TITLE
COOKIE_REFRESH always on 1h

### ISSUE THREAD

**Issue by LucaCre** [oauth2-proxy/oauth2-proxy#1950]
COOKIE_REFRESH always on 1h

## Expected Behavior
If I change the cookie-refresh value to for example "5m", I excpect that the expiration date will be 5min in the future.

## Current Behavior
But the value change doesn't show any effect. Exp time is still one hour.
The oauth2-proxy logs show that the config is made but it doesn't take any effect.

## Possible Solution

...

## Steps to Reproduce (for bugs)

OAUTH2_PROXY_COOKIE_REFRESH: '5m'
OAUTH2_PROXY_COOKIE_EXPIRE: '10m'

Logs: 
[oauthproxy.go:168] Cookie settings: name:_oauth2_proxy secure(https):false httponly:true expiry:10m0s domains: path:/ samesite: refresh:after 5m0s

Provider: OIDC

I'm using alpha config.

## Context

The refresh token will not be triggered after this hour past. Only if I reload the browser. Sincerely I'm not sure if this is a default behaviour?

## Your Environment

- Version used: 7.3.0

**Comment by miguelborges99** [comment]
Maybe some logs could help to understand if it is really an issue in oauth2 proxy.
Do you see any logs like this "Refreshing session - User xxxx; SessionAge: xxxx" occurring periodically ?
Do you receive a refresh token when you authenticate? If you do not receive a refresh token, then refresh will not occur.

**Comment by LucaCre** [comment]
I just see refresh_token:true, but no other logs regarding a refresh like SessionAge or others. 
So if the Exp time 1h shouldn't I have any log entry after an hour ?

**Comment by miguelborges99** [comment]
It should at least the log "Refreshing session - User". Do you have any errors in the logs?

**Comment by LucaCre** [comment]
No error and no entry like "Refreshing session - User"..

**Comment by miguelborges99** [comment]
If you try to authenticate, then do these logs occur? Or, nothing is printed in the logs?

**Comment by miguelborges99** [comment]
Looking into the code, it seems that refresh is executed only if you do a request. Oauth2-proxy is not actively/periodically refreshing the sessions. Only when user performs a request, then oauth2-proxy will look into your session and refresh it, if it is the case. This seems to be how oauth2-proxy is working.

**Comment by JoelSpeed** [comment]
OAuth2 Proxy typically doesn't track the sessions that are active, it evaluates them when they are loaded from the request. When it detects a session that has expired, but is refreshable, it will do so before returning the response. So if you wait for an hour, then hit the proxy, does it continue to work? If it does, you should see a log suggesting that things have been refreshed

**Comment by LucaCre** [comment]
Ok, I solved my problem. Maybe I didn't explained it well. I removed the OAUTH2_PROXY_COOKIE_REFRESH from the configuration. And set the OAUTH2_PROXY_COOKIE_EXPIRE to 1h. So after an hour the user have to login again. This was it what I actually wanted to achieve. I didn't understood that the OAUTH2_PROXY_COOKIE_REFRESH newer let the cookie expire even after  the time set by OAUTH2_PROXY_COOKIE_EXPIRE.
